File: android/src/main/kotlin/com/zeticai/zetic_mlange_flutter/wrapper/feature/yolov8/YOLOResultMapper.kt
package com.zeticai.zetic_mlange_flutter.wrapper.feature.yolov8

import com.zeticai.mlange.feature.entity.Box
import com.zeticai.mlange.feature.yolov8.YoloObject
import com.zeticai.mlange.feature.yolov8.YoloResult

object YOLOResultMapper {
    fun YoloResult.toMap(): Map<String, Any> {
        return mapOf(
            "value" to value.map { it.toMap() },
        )
    }

    fun YoloObject.toMap(): Map<String, Any> {
        return mapOf(
            "classId" to classId,
            "confidence" to confidence,
            "box" to box.toMap(),
        )
    }

    fun Box.toMap(): Map<String, Any> {
        return mapOf(
            "x" to xMin,
            "y" to yMin,
            "width" to (xMax - xMin),
            "height" to (yMax - yMin),
        )
    }
}

File: ios/Classes/wrapper/YOLOResultMapper.swift
import Foundation
import ZeticMLange

public struct YOLOv8ResultMapper {
    public static func toMap(_ result: YOLOv8Result) -> [String: Any] {
        let x     = result.box[0]
        let y     = result.box[1]
        let width = result.box[2]
        let height = result.box[3]
        
        let boxDictionary: [String: Int32] = [
            "x": x,
            "y": y,
            "width": width,
            "height": height
        ]
        
        return [
            "classId": result.classId,
            "confidence": result.confidence,
            "box": boxDictionary
        ]
    }
    public static func toMap(_ results: [YOLOv8Result]) -> [String: Any] {
        let mappedArray = results.map { toMap($0) }
        return ["value": mappedArray]
    }
}
